helix-term/src/ui/picker.rs

```diff
@@ -317,6 +321,26 @@ impl<T: Item> Picker<T> {
             |_editor: &mut Context, _pattern: &str, _event: PromptEvent| {},
         );
 
+        let n = options
+            .first()
+            .map(|option| option.row(&editor_data).cells.len())
+            .unwrap_or_default();
```

Comment:
I don't quite get what is this `n` for, is it the number of cells in the first row? Why do we only get the first row cells count?

Reply:
Yep `n` is the number of cells in the row, and we only take the first row on the assumption that all rows will have the same number of cells. What this block of the code does is calculate the maximum width that each column would need based on the longest item in that particular column. I copied the code from the implementation of [`Menu::recalculate_size()`](https://github.com/helix-editor/helix/blob/62fdcc8cbf7fd7496de2d17b6540b072f41c3344/helix-term/src/ui/menu.rs#L151-L187).